packages/babel-parser/src/tokenizer/index.js

```diff
@@ -790,6 +790,40 @@ export default class Tokenizer extends ParserErrors {
     }
   }
 
+  readToken_at(): void {
+    // Increase past @
+    ++this.state.pos;
+    const next = this.input.charCodeAt(this.state.pos);
+
+    if (isIdentifierStart(next)) {
+      // Read the next word with a lookahead state that we'll
+      // throw away afterwards. Cheaper than doing a full lookahead,
+      // since we know that we're trying to read a word/identifier.
+      const old = this.state;
+      this.state = this.createLookaheadState(old);
+
+      const word = this.readWord1(next);
+      const wordNext = this.input.charCodeAt(this.state.pos);
+      const containsEsc = this.state.containsEsc;
+
+      // Replace the previous state
+      this.state = old;
+
+      if (wordNext === charCodes.colon) {
+        if (word !== "init" || containsEsc) {
```

Comment:
If `containsEsc` we could raise `Errors.InvalidEscapedReservedWord`, which is a bit more descriptive.